test/e2e/pkg/suite_test.go

```diff
@@ -0,0 +1,103 @@
+package e2e_test
+
+import (
+	"crypto/tls"
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+)
+
+var (
+	apiServerAddress string
+	kubeconfig       string
+	consumer_id      string
+	kubeClient       *kubernetes.Clientset
+)
+
+func TestE2E(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "End-to-End Test Suite")
+}
+
+func init() {
+	klog.SetOutput(GinkgoWriter)
+	klog.InitFlags(nil)
+	flag.StringVar(&apiServerAddress, "api-server", "", "Maestro API server address")
+	flag.StringVar(&consumer_id, "consumer_id", "", "Connsumer ID is used to identify the consumer")
+	flag.StringVar(&kubeconfig, "kubeconfig", "", "Path to kubeconfig file")
+}
+
+var _ = BeforeSuite(func() {
+	// validate the maestro api server is running
+	_, err := sendHTTPRequest(http.MethodGet, apiServerAddress+"/api/maestro", nil)
+	if err != nil {
+		panic(fmt.Sprintf("failed to connect to maestro api server: %v", err))
+	}
+
+	// validate the kubeconfig file
+	restConfig, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		panic(fmt.Sprintf("failed to build kubeconfig: %v", err))
+	}
+	kubeClient, err = kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		panic(fmt.Sprintf("failed to create kube client: %v", err))
+	}
+
+	// validate the consumer_id
+	if consumer_id == "" {
+		panic("consumer_id is not provided")
+	}
+})
+
+var _ = AfterSuite(func() {
+	// later...
+})
+
+func buildHTTPClient(apiServerAddress string) *http.Client {
```

Comment:
this project generates a REST client from openapi.  why not use that?  It only requires an authenticated context (and URL) to work.  

I think we should test our client rather than create another.

Reply:
Done